### PR TITLE
Create contribution_guide

### PR DESCRIPTION
**Pull request by hangfei**
Added contribution guide.

### CODE REVIEW

**Review by xiaoyongzhu — APPROVED** [review]

File: docs/dev_guide/contribution_guide.MD [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+If you are willing to learn, no problem at all!
+
+You can ping your questions in the community Slack channel and Feathr developers will give you more guidance. For example:
+* "I am not familiar with Scala, but I want to start to contirbute to Scala code. Can someone give me some guidance?"
+* "I am not sure how to setup the cluster and test my code. Can someone help me out?"
+
+## My pull request(PR) requires testing against the database or cluster that I dont' have, how can i fix this?
+(We are still working on this. Stay tuned.)
+
+## Technical Architecutre
```
Comment (blrchen) [inline docs/dev_guide/contribution_guide.MD:33]: Architecutre => Architecture

File: docs/dev_guide/contribution_guide.MD [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+# I am interested, how can I start?
+If you are new to this project, we recommend start with [`good-first-issue`](https://github.com/linkedin/feathr/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22)
+
+If you are familiar with this project, you can just start to pick up our github issues.
+
+## I am not familiar with XYZ, can I still contribute?
+If you are willing to learn, no problem at all!
+
+You can ping your questions in the community Slack channel and Feathr developers will give you more guidance. For example:
+* "I am not familiar with Scala, but I want to start to contirbute to Scala code. Can someone give me some guidance?"
```
Comment (blrchen) [inline docs/dev_guide/contribution_guide.MD:27]: >contirbute [](http://example.com/codeflow?start=56&length=10)

contirbute => contribute

File: docs/dev_guide/contribution_guide.MD [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+* Computation Engine
+  * The computation engine that execute the actual feature join and generation work. Mostly in Scala and Spark.
+* Feature Registry API Layer
+  * The storage layer supports SQL, Purview(Atlas).
+  * The API layer is in Python(FAST API)
+* Feature Registry Web UI layer
+  * The Web UI for feature registry. Written in React with a few UI frameworks.
+
+# Developer Guide
+See [`feathr/docs/dev_guide`](../)
```
Comment (blrchen) [inline docs/dev_guide/contribution_guide.MD:46]: This link is broken

File: docs/dev_guide/contribution_guide.MD [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+You can ping your questions in the community Slack channel and Feathr developers will give you more guidance. For example:
+* "I am not familiar with Scala, but I want to start to contirbute to Scala code. Can someone give me some guidance?"
+* "I am not sure how to setup the cluster and test my code. Can someone help me out?"
+
+## My pull request(PR) requires testing against the database or cluster that I dont' have, how can i fix this?
+(We are still working on this. Stay tuned.)
+
+## Technical Architecutre
+Feathr contains multiple components so you can learn and contribute to various things:
+* Python Feathr Client
```
Comment (blrchen) [inline docs/dev_guide/contribution_guide.MD:35]: Suggest using a table to make reading easiler

**Review by windoze — DISMISSED** [review]

**Review by windoze — APPROVED** [review]

**Review by xiaoyongzhu — APPROVED** [review]

**Pull request merged**